src/ClimaAtmos/Dycore/src/VanillaAtmosDiscretizations_kernels.jl

```diff
@@ -400,41 +484,80 @@ function volumerhs!(::Val{2}, ::Val{N}, ::Val{nmoist}, ::Val{ntrace},
                     rhs::Array, Q, grad, vgeo, gravity, viscosity, D,
                     elems) where {N, nmoist, ntrace}
   DFloat = eltype(Q)
-
-  nvar = _nstate + nmoist + ntrace
-  ngrad = _nstategrad + 3nmoist
+  nvar   = _nstate + nmoist + ntrace
+  ngrad  = _nstategrad + 3nmoist
+  dim    = 2
 
   Nq = N + 1
 
   nelem = size(Q)[end]
 
-  Q = reshape(Q, Nq, Nq, nvar, nelem)
+  Q    = reshape(Q, Nq, Nq, nvar, nelem)
   grad = reshape(grad, Nq, Nq, ngrad, nelem)
-  rhs = reshape(rhs, Nq, Nq, nvar, nelem)
+  rhs  = reshape(rhs, Nq, Nq, nvar, nelem)
   vgeo = reshape(vgeo, Nq, Nq, _nvgeo, nelem)
 
   s_F = Array{DFloat}(undef, Nq, Nq, _nstate)
   s_G = Array{DFloat}(undef, Nq, Nq, _nstate)
   l_u = Array{DFloat}(undef, Nq, Nq)
   l_v = Array{DFloat}(undef, Nq, Nq)
 
+  q_m = zeros(DFloat, max(3, nmoist))
+    
   @inbounds for e in elems
+
+    delta = Grids.compute_anisotropic_grid_factor(dim, Nq, vgeo, e)
```

Comment:
Since grid anisotropy doesn't change during the simulation, its possible that this should be stored in an array. Also, for a warped and skewed element does it really make sense that this is constant throughout the element?